### PR TITLE
web-ui: Add tests for client

### PR DESCRIPTION
**Pull request by bhaskarmelkani**
Add tests for frontend and improve coverage.
Related #212

### CODE REVIEW

File: web-ui/src/client/app/components/__tests__/violations.test.jsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+      const component = shallow(<ViolationsResult
+          pending={false}
+          complete={true}
+          errorMsgText={'Server Error'}
+          violations={[]}
+          successMsgTitle=''
+          successMsgText=''
+        />);
+
+      expect(component.find('If').at(2).dive().find('Msg').length).toEqual(1);
```
Comment (fmueller) [inline web-ui/src/client/app/components/__tests__/violations.test.jsx]: This looks very tightly coupled to production code. These tests should not break when UI template is changed a bit. Especially `at(2)` looks strange to me - but I don't know React in detail. So, maybe I'm wrong.

Comment (rbarilani) [inline web-ui/src/client/app/components/__tests__/violations.test.jsx]: @fmueller Nope you absolutely right. We were not happy with it too.
We will use some specific ui selectors to allow moving the related DOM component without breaking the tests (...at least we hope )

Comment (fmueller) [inline web-ui/src/client/app/components/__tests__/violations.test.jsx]: @rbarilani Is it possible to use data-attributes which are only used for tests to identity correct elements?

Comment (rbarilani) [inline web-ui/src/client/app/components/__tests__/violations.test.jsx]: @fmueller maybe

Comment (bhaskarmelkani) [inline web-ui/src/client/app/components/__tests__/violations.test.jsx]: @fmueller @rbarilani I have added data-attributes for the same.

**Comment by bhaskarmelkani** [comment]
👍

File: web-ui/src/client/app/components/__tests__/violations.test.jsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+      const component = shallow(<ViolationsResult
+          pending={false}
+          complete={true}
+          errorMsgText={'Server Error'}
+          violations={[]}
+          successMsgTitle=''
+          successMsgText=''
+        />);
+
+      expect(component.find('If[dataTestId="if-error"]').dive().find('Msg').length).toEqual(1);
```
Comment (rbarilani) [inline web-ui/src/client/app/components/__tests__/violations.test.jsx]: @bhaskarmelkani Can you assign selectors to variables if they are going to be reused?

Something like:

```
const IFErrorSelector = 'If[dataTestId="if-error"]';
const ifError = component.find(IFErrorSelector);
const Msg = iFError.dive().find('Msg');
expect(Msg.prop('type')).toEqual('error');
expect(Msg.prop('title').toEqual('ERROR');
// ...
```
Same refactor required on other parts of the code where selector are repeated.

File: web-ui/src/client/app/components/violations.jsx [hunk 3]
```diff
@@ -66,16 +66,16 @@ function ViolationPaths(props) {
 function ViolationsResult(props){
   return (
     <div className="violations-result">
-      <If test={() => props.pending}>
+      <If dataTestId="if-loading" test={() => props.pending}>
```
Comment (rbarilani) [inline web-ui/src/client/app/components/violations.jsx]: @bhaskarmelkani can you put the test property as the last property?

Comment (bhaskarmelkani) [inline web-ui/src/client/app/components/violations.jsx]: Sure

File: web-ui/src/client/app/components/__tests__/violations.test.jsx (hunk 2 above)
Comment (bhaskarmelkani) [inline web-ui/src/client/app/components/__tests__/violations.test.jsx]: Sure

File: web-ui/src/client/app/components/__tests__/form.test.jsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      target: {
+        value: 'foo'
+      }
+    }
+    input.simulate('change', event);
+    expect(component.state('inputValue')).toEqual('foo');
+
+    const form = component.find('form');
+    form.simulate('submit', {preventDefault() {}});
+    form.simulate('submit', {preventDefault() {}});
```
Comment (bhaskarmelkani) [inline web-ui/src/client/app/components/__tests__/form.test.jsx]: It is as there are two responses of the event handler. 1. Resolving and rejecting the promise.

Comment (bhaskarmelkani) [inline web-ui/src/client/app/components/__tests__/form.test.jsx]: But maybe, I will refractor it.

**Comment by bhaskarmelkani** [comment]
👍

File: web-ui/src/client/app/components/__tests__/form.test.jsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+          value: 'URL_WITH_AN_ERROR'
+        }
+      }
+      input.simulate('change', event);
+      form = component.find('form');
+
+      form.simulate('submit', {preventDefault() {}});
+      expect(getApiViolations.mock.calls[0][0]).toBe('URL_WITH_AN_ERROR');
+    });
+    afterEach(() => {
```
Comment (rbarilani) [inline web-ui/src/client/app/components/__tests__/form.test.jsx]: @bhaskarmelkani move afterEach block after beforeEach

File: web-ui/src/client/app/components/violations.jsx (hunk 3 above)
Comment (bhaskarmelkani) [inline web-ui/src/client/app/components/violations.jsx]: Done

File: web-ui/src/client/app/components/__tests__/violations.test.jsx (hunk 2 above)
Comment (bhaskarmelkani) [inline web-ui/src/client/app/components/__tests__/violations.test.jsx]: Done

**Comment by bhaskarmelkani** [comment]
👍

File: web-ui/src/client/app/components/__tests__/form.test.jsx (hunk 5 above)
Comment (bhaskarmelkani) [inline web-ui/src/client/app/components/__tests__/form.test.jsx]: Done

**Comment by bhaskarmelkani** [comment]
👍

**Comment by rbarilani** [comment]
👍

**Pull request merged**